lib/internal/url.js

```diff
@@ -535,14 +534,16 @@ ObjectDefineProperties(URLSearchParams.prototype, {
 });
 
 function isURL(self) {
```

Comment:
We should probably comment this function explaining why we checking those properties
```suggestion
/**
 * Checks if a value has the shape of a WHATWG URL object.
 *
 * Using a symbol or instanceof would not be able to recognize URL objects
 * coming from other implementations (e.g. in Electron), so instead we are
 * checking some well known properties for a lack of a better test.
 *
 * @param {*} self
 * @returns {asserts value is URL}
 */
function isURL(self) {
```

Reply:
I agree. Updated pull request. Thanks for the feedback.